### PR TITLE
[trie] Refactor trie ops related code

### PR DESCRIPTION
**Pull request by shreyan-gupta**
Pure refactor, after this it would be easier to determine what code belongs to generics (all fall under ops/ directory) and which code is specific to memtries and normal tries.

### CODE REVIEW

File: core/store/src/trie/node_storage.rs [hunk 1]
```diff
@@ -1,24 +1,69 @@
...
+impl UpdatedTrieStorageNodeWithSize {
+    pub fn from_raw_trie_node_with_size(node: RawTrieNodeWithSize) -> Self {
+        Self {
+            node: UpdatedTrieStorageNode::from_raw_trie_node(node.node),
+            memory_usage: node.memory_usage,
+        }
+    }
+}
+
 pub(crate) struct NodesStorage<'a> {
```
Comment (Longarithm) [inline core/store/src/trie/node_storage.rs:60]: As we are on it:
* let's rename `NodesStorage` to `TrieStorageUpdate` to make its purpose more clear (and rename the file as well),
* let's make `flatten_nodes` and `flatten_value` methods of `TrieStorageUpdate`, it's not actually a `Trie` method. Moreover, `Trie` is a field of `TrieStorageUpdate`.

**Review by Longarithm — APPROVED** [review]

File: core/store/src/trie/node_storage.rs (hunk 1 above)
Comment (shreyan-gupta) [reply core/store/src/trie/node_storage.rs:60]: Done

File: core/store/src/trie/ops/insert_delete.rs [hunk 2]
```diff
@@ -0,0 +1,457 @@
...
+            GenericUpdatedTrieNodeWithSize { node: new_node, memory_usage: new_memory_usage },
+        );
+    }
+
+    /// Insertion logic. We descend from the root down to whatever node corresponds to
+    /// the inserted value. We would need to split, modify, or transform nodes along
+    /// the way to achieve that. This takes care of refcounting changes for existing
+    /// nodes as well as values, but will not yet increment refcount for any newly
+    /// created nodes - that's done at the end.
+    fn generic_insert(
```
Comment (Trisfald) [inline core/store/src/trie/ops/insert_delete.rs:39]: Personal take: on `GenericTrieUpdateInsertDelete::generic_insert` and similar method we could drop one occurrence of "generic" or even both. 
I'd be ok with a shorter `TrieUpdateInsertDelete::insert`

File: core/store/src/trie/node_storage.rs (hunk 1 above)
Comment (Trisfald) [inline core/store/src/trie/node_storage.rs]: Would it be possible to add a short comment to explain the usage and/or purpose of this struct?

Comment (shreyan-gupta) [reply core/store/src/trie/node_storage.rs]: Will do in a follow up PR

**Pull request merged**